firmware/library/L3_HAL/st7066u_test.cpp

```diff
@@ -0,0 +1,266 @@
+// Tests for the St7066u Parallel LCD Driver class.
+#include "L3_HAL/st7066u.hpp"
+#include "L5_Testing/testing_frameworks.hpp"
+
+TEST_CASE("Testing St7066u Parallel LCD Driver", "[st7066u]")
+{
+    class GpioMock : public Gpio
+    {
+     public:
+        constexpr GpioMock(uint8_t port, uint8_t pin) : Gpio(port, pin) {}
+        void SetHigh() final
+        {
+            gpio_base[kPort]->PIN |= (1 << kPin);
+        }
+        void SetLow() final
+        {
+            gpio_base[kPort]->PIN &= ~(1 << kPin);
+        }
+    };
+
+    // Setup local LPC_GPIO1 and LPC_GPIO2 port registers for simulation.
+    LPC_GPIO_TypeDef local_gpio[2];
+    memset(&local_gpio[0], 0, sizeof(local_gpio[0]));
+    memset(&local_gpio[1], 0, sizeof(local_gpio[1]));
+    // Substitute LPC_GPIO1 and LPC_GPIO2 with the local test ports
+    Gpio::gpio_base[1] = &local_gpio[0];
+    Gpio::gpio_base[2] = &local_gpio[1];
+
+    // GPIO pin numbers for control pins.
+    constexpr uint8_t kRSPin = 30;
+    constexpr uint8_t kRWPin = 29;
+    constexpr uint8_t kEPin  = 28;
+    constexpr uint8_t kD7Pin = 7;
+    constexpr uint8_t kD6Pin = 6;
+    constexpr uint8_t kD5Pin = 5;
+    constexpr uint8_t kD4Pin = 4;
+    constexpr uint8_t kD3Pin = 3;
+    constexpr uint8_t kD2Pin = 2;
+    constexpr uint8_t kD1Pin = 1;
+    constexpr uint8_t kD0Pin = 0;
+
+    constexpr uint8_t kCommandWrite   = 0;
+    constexpr uint8_t kDataWrite      = 1;
+    constexpr uint8_t kWriteOperation = 0;
+
+    // Mock GPIO control pinsÎ
+    GpioMock rs = GpioMock(1, kRSPin);  // RS: register select pin
+    GpioMock rw = GpioMock(1, kRWPin);  // RW: read/write pin
+    GpioMock e  = GpioMock(1, kEPin);   // E:  chip enable pin
+    GpioMock d7 = GpioMock(2, kD7Pin);  // D7-D0: Parallel Data pins
+    GpioMock d6 = GpioMock(2, kD6Pin);
+    GpioMock d5 = GpioMock(2, kD5Pin);
+    GpioMock d4 = GpioMock(2, kD4Pin);
+    GpioMock d3 = GpioMock(2, kD3Pin);
+    GpioMock d2 = GpioMock(2, kD2Pin);
+    GpioMock d1 = GpioMock(2, kD1Pin);
+    GpioMock d0 = GpioMock(2, kD0Pin);
+
```

Comment:
Please move to using fakeit rather than making your own mock class to conform with the rest of the tests.